nomad/fsm_test.go

```diff
@@ -2847,6 +2847,126 @@ func TestFSM_ReconcileSummaries(t *testing.T) {
 	}
 }
 
+func TestFSM_ReconcileVersionedSummary(t *testing.T) {
+	logger := testlog.HCLogger(t)
+	logger.Info("+++++")
+
+	ci.Parallel(t)
+	// Add some state
+	fsm := testFSM(t)
+	state := fsm.State()
+
+	// Add a node
+	node := mock.Node()
+	require.NoError(t, state.UpsertNode(structs.MsgTypeTestSetup, 800, node))
+
+	// Make a job
+	job1 := mock.Job()
+	job1.TaskGroups[0].Tasks[0].Resources.CPU = 5000
+
+	// // make a job which can make partial progress
+	alloc := mock.Alloc()
+	alloc.NodeID = node.ID
+	// Not sure why, but have to set both .Job and .JobID for this alloc to show up in the job summary??
```

Comment:
If anybody knows why this happens, please lmk! If I only set .Job or .JobID, the allocation is uncounted in .summary or .versionedSummary